binder/plugins/models/html_field.py

```diff
@@ -0,0 +1,80 @@
+from django.db.models import TextField
+from html.parser import HTMLParser
+from django.core import exceptions
+
+
+def link_validator(tag, attribute_name, attribute_value):
+	if not attribute_value.startswith('http://') and not attribute_value.startswith('https://'):
+		raise exceptions.ValidationError(
+			'Link is not valid',
+			code='invalid_tag',
+			params={
+				'tag': tag,
+			},
+		)
+
+
+class HtmlValidator(HTMLParser):
+	allowed_tags = [
+		# General setup
+		'p', 'br',
+		# Headers
+		'h1', 'h2', 'h3', 'h4', 'h5', 'h6', 'h7',
+
+		# text decoration
+		'b', 'strong', 'i', 'em', 'u',
+		# Lists
+		'ol', 'ul', 'li',
+
+		# Special
+		'a',
+	]
+
+	allowed_attributes = {
+		'a': ['href', 'rel', 'target']
+	}
+
+	special_validators = {
+		('a', 'href'): link_validator
+	}
+
+	error_messages = {
+		'invalid_tag': 'Tag %(tag)s is not allowed',
+		'invalid_attribute': 'Attribute %(attribute)s not allowed for tag %(tag)s'
+	}
+
+	def handle_starttag(self, tag: str, attrs: list) -> None:
+		if tag not in self.allowed_tags:
```

Comment:
Is it not best to first gather all errors in a list/dict and then return all at once, rather then instantly raising on the first error, while the input in question may contain validation errors for both cases.

Reply:
Ideally yes, but I don't see how more than one validationerrors can be thrown

Reply:
In the current functionality indeed, however in the future this may be needed. Plus in general imho it is best to keep with one design pattern for handling any data cleaning within binder. In that case the pattern where we return all errors at once instead of one error at a time upon save would result in the better user experience i would think.